framework/core/Project.pm

```diff
@@ -982,6 +987,20 @@ sub contains_version_id {
     return $self->{_vcs}->contains_version_id($vid);
 }
 
+=pod
+=item B<checkout_id> C<checkout_id(version [, work_dir])>
```

Comment:
Can you change `version` to `revision_id`?

Also, `work_dir` should be `prog_root` in this routine.

Reply:
`version` should really be a `vid` right?  VCS::checkout_vid accepts a `vid` and a `work_dir` (which should be named `prog_root`)